src/tokens/saga.ts

```diff
@@ -233,25 +238,47 @@ interface UserBalancesResponse {
 export async function fetchTokenBalancesForAddress(
   address: string
 ): Promise<FetchedTokenBalance[]> {
-  const response = await apolloClient.query<UserBalancesResponse, { address: string }>({
-    query: gql`
-      query FetchUserBalances($address: Address!) {
-        userBalances(address: $address) {
-          balances {
-            tokenAddress
-            balance
+  const chainsToFetch = getFeatureGate(StatsigFeatureGates.FETCH_MULTI_CHAIN_BALANCES)
```

Comment:
Reading the PR title, I assumed this PR was just updating the token metadata source. This seems like a big change that should also go in the PR title. Can we update the title to reflect that? (Would be ideal if fetching multi chain balances is in its own PR)

Reply:
Yeah, title has been updated; I agree that this likely should have been split out into separate PRs; this has gotten a bit bigger than anticipated.